lib/components/javascript.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-undef */
+// this function will be stringified as part of the generated types and is not supposed to be used internally
+const proxyAccessFunction = function (namespace, name, opts = {}) {
+    const { target, customProps } = { target: {}, customProps: [], ...opts }
+    const fq = namespace !== '' ? `${namespace}.${name}` : name
+    let _entity
+    return new Proxy(target, {
+        get: function (target, prop) {
+            // we already know the name so we skip the cds.entities proxy access
+            if (prop === 'name') return fq
+            // custom properties access on 'target' as well as cached _entity property access goes here
+            if (Object.prototype.hasOwnProperty.call(target, prop)) return target[prop]
```

Comment:
```suggestion
            if (Object.hasOwn(target, prop)) return target[prop]
``` 
or is there a specific reason to go via the prototype here?

Reply:
1) I did not know about `hasOwn`
2) I wrote the code in typescript and changed it to `prototype` because of the eslint rule https://eslint.org/docs/latest/rules/no-prototype-builtins